### PR TITLE
Add type file and missing files

### PR DESCRIPTION
**Pull request by berkbaski**
Hey!

I want to do two more fixes.

### 1. Updating the JavaScript samples in the README.md file to TypeScript.
In the examples on the README.md file, the types are given as characters and comment lines. I don't know how to update this.

Like this
```javascript
spriteObject.setSprite({
	sheet: Image.createFromFile("images://spritesheet.png"),
	frameX: << #frames in axis-X >>, // Number
	frameY: << #frames in axis-Y >>, // Number
	frameCount: << #frames in the sheet >> // Number
});
```

### 2. I want to make the parameter in the constructor optional if necessary. 
But I think incoming parameters pass to the Image object. I wasn't sure if I should do this.

I have time for suggestions.

### CODE REVIEW

**Comment by furkanarabaci** [comment]
For 1#, I think it is better to leave the code snippet as is. Only change requires to import and it will be enough.

For 2#, if the parameters are passed to image object, then it is better to have them as optional.

Btw when I talk about constructor, I mean this:
`const spriteView = new SpriteView({// Parameters here});`

You should also have copy-paste ready snippet like this:

```
import SpriteView from 'sf-extension-spriteview';
cons spriteView = new SpriteView(); // Constructor parameters should be typed as optional, if there is any
spriteView.setSprite({
        sheet: Image.createFromFile("images://spritesheet.png"),
	frameX: 30, // This needs a brief explanation in documents
	frameY: 30, // This needs a brief explanation in documents
	frameCount: 24 // This needs a brief explanation in documents
});
```

The parameter in setSprite shouldn't be optional. I think every property in options parameter is required.

**Comment by berkbaski** [comment]
Okay, thank you for your suggestions.

1# I updated with your suggestion. By the way, I added the optional fields.
2# I tested your suggestion by IDE. Cool, didn't break! But not displaying if you didn't set the width, height fields. I should give a default value to these fields maybe?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * Image for the frame
+     */
+    sheet: Image,
+    /**
+     * X Position of the window
+     */
+    frameX: number,
+    /**
+     * Y Position of the window
```
Comment (berkbaski) [inline index.d.ts]: Frame Y count
Distinct image count on Y axis

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    imageFillType: ImageView.FillType;
+}
+
+export type SetSpriteOptions = {
+    /**
+     * Image for the frame
+     */
+    sheet: Image,
+    /**
+     * X Position of the window
```
Comment (berkbaski) [inline index.d.ts]: Frame X count
Distinct image count on X axis

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+     */
+    frameX: number,
+    /**
+     * Y Position of the window
+     */
+    frameY: number,
+    /**
+     * Frame count of the image
+     */
+    frameCount: number
```
Comment (berkbaski) [inline index.d.ts]: Kill it!

Comment (berkbaski) [reply index.d.ts]: @furkanarabaci have an idea here? Sometimes I want to spoil the animation, maybe? I tested it and it shows as many frames as given.

Comment (furkanarabaci) [reply index.d.ts]: Make frameCount optional with default value to x*y

**Review by furkanarabaci — CHANGES_REQUESTED** [review]
FIX THIS THING OR RETURN TO YOUR GODDAMN VILLAGE!

File: index.d.ts (hunk 3 above)
Comment (berkbaski) [reply index.d.ts]: Okay, did it!

**Pull request merged**